File: Practice_Rust/L4/l4.cpp
#include <bits/stdc++.h>
using namespace std;
int main() {
	int a = 7; // a = 0, even in this case Value 3 gets printed
	if (a = 3) { // it just assigns 3 as soon as you write a = 3 in the parentheses
		cout << "Value 3" << '\n';
	}
	else {
		cout << "Not 3" << '\n';
	}
	return 0;
}

File: Practice_Rust/L4/l4.rs
fn main() {
	let mut a = 7;
	if a == 3 { // in rust you cannot write a = 3 here, so I will write a c++ code to explain
		println!("Value 3");
	}
	else {
		println!("Not 3");
	}
	// here we just get Value 3 as the output. NOTE: Even a = 0 works and will still output Value 3.
}